diff --git a/build_scripts/build.sh b/build_scripts/build.sh
index 7c4b2eb..e5fa216 100755
--- a/build_scripts/build.sh
+++ b/build_scripts/build.sh
@@ -128,18 +128,18 @@ if [ "$NODE_VERSION" != "v$EXPECTED_NODE_VERSION" ]; then
 fi
 echo -e "\033[32m[OHOS INFO] Node.js version check passed!\033[0m"
 # HUANGLONG begin
-#npm config set registry https://repo.huaweicloud.com/repository/npm/
-#npm config set @ohos:registry https://repo.harmonyos.com/npm/
-#npm config set strict-ssl false
-#npm config set lockfile false
-#cat $HOME/.npmrc | grep 'lockfile=false' > /dev/null || echo 'lockfile=false' >> $HOME/.npmrc > /dev/null
+npm config set registry https://repo.huaweicloud.com/repository/npm/
+npm config set @ohos:registry https://repo.harmonyos.com/npm/
+npm config set strict-ssl false
+npm config set lockfile false
+cat $HOME/.npmrc | grep 'lockfile=false' > /dev/null || echo 'lockfile=false' >> $HOME/.npmrc > /dev/null
 # HUANGLONG end
 function init_ohpm() {
   TOOLS_INSTALL_DIR="${SOURCE_ROOT_DIR}/prebuilts/build-tools/common"
   pushd ${TOOLS_INSTALL_DIR} > /dev/null
     if [[ ! -f "${TOOLS_INSTALL_DIR}/oh-command-line-tools/ohpm/bin/ohpm" ]]; then
       echo "[OHOS INFO] download oh-command-line-tools"
-      wget https://repo.huaweicloud.com/harmonyos/ohpm/5.0.2/oh-command-line-tools-20240715.zip -O ohcommandline-tools-linux.zip
+      wget --no-check-certificate https://repo.huaweicloud.com/harmonyos/ohpm/5.0.2/oh-command-line-tools-20240715.zip -O ohcommandline-tools-linux.zip
       unzip ohcommandline-tools-linux.zip
     fi
     OHPM_HOME=${TOOLS_INSTALL_DIR}/oh-command-line-tools/ohpm/bin
@@ -172,7 +172,7 @@ function init_ohpm() {
 if [[ "$*" != *ohos-sdk* ]]; then
   echo "[OHOS INFO] Ohpm initialization started..."
 # HUANGLONG begin
-  #init_ohpm
+  init_ohpm
 # HUANGLONG end
   if [[ "$?" -ne 0 ]]; then
     echo -e "\033[31m[OHOS ERROR] ohpm initialization failed!\033[0m"
